### PR TITLE
New release notes format

### PR DESCRIPTION
**Pull request by yaakovi**
## Status
In Progress - Not fully tested yet

## Description
moving from release notes files to change log files.
change log will record all notable changes made to a content artifact.
same as release notes, each file will have it's own change log.

workflow for developers:
* whenever changing a file:
- if not already exists, add a change log with the same name as the relevant yml/json file with the suffix `_changelog.md`. PAY ATTENTION, file needs to be at the same directory as the changed file.
- if exist, always add your notes to the beginning of the file. this should be enforce at the PR level by the reviewer (and in the future, the file validation step).

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Dependencies
Mention the dependencies of the entity you changed as given from the precommit hooks in checkboxes, and tick after tested them.
- [ ] Dependency 1
- [ ] Dependency 2
- [ ] Dependency 3

## Additional changes
converted every release notes file to change log file - not done yet.

### CODE REVIEW

File: Tests/test_utils.py [hunk 1]
```diff
@@ -142,7 +142,32 @@ def str2bool(v):
...
+        return None
+
+    if re.match(CONTENT_RELEASE_TAG_REGEX, rn):
+        # in case rn already starts with a release tag - they considered invalid
+        return None
+
+    new_rn = re.findall(RELEASE_NOTES_REGEX, rn)
+    if new_rn:
+        # get release notes up to release header
+        return new_rn[0]
```
Comment (yaakovi) [inline Tests/test_utils.py]: add `strip`

**Comment by yaakovi** [comment]
1) done
2) done
3) it will only run 2 sanity tests since no change will be done to content artifacts.
4) wrote a script for that, once the PR is approved, i'll commit the converted files (this needs to be done as close to the merge as possible to avoid conflicts in master)

File: release_notes_clear.py [hunk 2]
```diff
@@ -1,129 +1,76 @@
...
+    '.yml': yaml.safe_load,
+    '.json': json.load,
+}
 
 
-FILE_EXTRACTER_DICT = {
-    '*.yml': yml_remove_releaseNote_record,
-    '*.json': json_remove_releaseNote_record,
-}
+def get_changed_content_artifacts(modified_files, added_files):
```
Comment (Itay4) [inline release_notes_clear.py]:
```suggestion
def get_changed_content_entities(modified_files, added_files):
```

File: release_notes_clear.py [hunk 3]
```diff
@@ -1,129 +1,76 @@
...
+    return added_files.union([(file_path[1] if isinstance(file_path, tuple) else file_path)
```
Comment (Itay4) [inline release_notes_clear.py]: add comment

**Pull request merged**